src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -165,6 +172,15 @@ public TopicPartitionWriter(
       this.hiveUpdateFutures = hiveUpdateFutures;
       hivePartitions = new HashSet<>();
     }
+
+    if(rotateScheduleIntervalMs > 0) {
```

Comment:
Moving check into another class and use check result is very fragile for me. We could pass this properties to constructor, but `TopicPartitionWritter` designed to get his properties from config. For me it is ok, to keep this check in `TopicPartitionWritter`.